src/io/flutter/preview/PreviewArea.java

```diff
@@ -62,44 +66,75 @@
   public PreviewArea(Listener listener) {
     this.myListener = listener;
 
-    panel.addComponentListener(new ComponentAdapter() {
+    primaryLayer.addComponentListener(new ComponentAdapter() {
       @Override
       public void componentResized(ComponentEvent e) {
-        final int width = panel.getWidth() - 2 * BORDER_WITH;
-        final int height = panel.getHeight() - 2 * BORDER_WITH;
+        final int width = primaryLayer.getWidth() - 2 * BORDER_WITH;
+        final int height = primaryLayer.getHeight() - 2 * BORDER_WITH;
         listener.resized(width, height);
       }
     });
 
-    panel.setLayout(new BorderLayout());
+    primaryLayer.setLayout(new BorderLayout());
     clear();
 
+    // Layers must be transparent.
     handleLayer.setOpaque(false);
-    panel.addComponentListener(new ComponentAdapter() {
+
+    layeredPanel.add(primaryLayer, Integer.valueOf(0));
+    layeredPanel.add(handleLayer, Integer.valueOf(1));
+
+    // Layers must cover the whole root panel.
+    layeredPanel.addComponentListener(new ComponentAdapter() {
       @Override
       public void componentResized(ComponentEvent e) {
-        handleLayer.setBounds(0, 0, panel.getWidth(), panel.getHeight());
+        final int width = layeredPanel.getWidth();
+        final int height = layeredPanel.getHeight();
+        for (Component child : layeredPanel.getComponents()) {
+          child.setBounds(0, 0, width, height);
+        }
       }
     });
-    panel.add(handleLayer);
+  }
+
+  /**
+   * Return the Swing component of the area.
+   */
+  public JComponent getComponent() {
+    return layeredPanel;
   }
 
   public void clear() {
-    panel.removeAll();
-    panel.setLayout(new BorderLayout());
-    panel.add(new JLabel("Nothing to show", SwingConstants.CENTER), BorderLayout.CENTER);
-    panel.repaint();
+    primaryLayer.removeAll();
+    primaryLayer.setLayout(new BorderLayout());
+    primaryLayer.add(new JLabel("Nothing to show", SwingConstants.CENTER), BorderLayout.CENTER);
```

Comment:
Perhaps have text similar to:

<img width="646" alt="screen shot 2018-03-12 at 10 54 32 am" src="https://user-images.githubusercontent.com/1269969/37300682-cbfd6b6e-25e3-11e8-8ebd-ed66fb66494d.png">

Although, longer term, we'd want to iterate to help the user understand why there was no preview, and the steps they could take to make one available.

Reply:
+1